index.md

```diff
@@ -407,96 +418,127 @@ The third, fourth, and fifth items are there to make it easy for other people to
-and distinguished clearly from the other scientific code by a name such as "runall".
+For a small project with one main output,
+a single controller script should be placed in the main `src` directory
+and distinguished clearly by a name such as "runall".
 
-The controller script should be thought of simply as the "glue" that holds the analysis together and allows a single command,
+The controller script is the glue that holds the analysis together
+and allows a single command,
 such as `python runall.py`,
-to re-run the entire analysis, from start to finish.
-These scripts should be short, no more than 100-200 lines at most, and be very easy to understand.
-If this script becomes longer than this,
+to re-run the entire analysis from start to finish.
+These scripts should be short (no more than 100-200 lines at most),
+and be very easy to understand:
+in particular,
+while they may contain loops (to process multiple data files or sweep across parameter ranges)
+they should contain few if any conditional statements
+or new function definitions.
+If a control script becomes longer or more complicated than this,
 or begins to include code that would require a new collaborator more than a minute or two to understand,
-these portions of the code should be moved out of the controller script and into other core analysis files in this directory.
-
-The `results` directory will also generally require additional structure for all but the simplest projects.
-At a minimum, intermediate files such as cleaned data, statistical tables, and final publication-ready figures or tables
-should be separated clearly by file naming conventions or placed into different subdirectories.
-
-As previously noted,
-in an ideal project a controller script should be able to create all of the results found in the `results` directory automatically,
-using the contents of `data`, `src`, and/or `bin`, with no manual human intervention.
-This helps to ensure both provenance,
-so that results can always be associated with the upstream files that generated them,
-and reproducibility, so that you or others can recreate the outputs of any analysis.
-If this goal is achieved,
-then the contents of the `results` directory do not need to be placed under version control (see below),
-as they do not contain any unique information.
+those portions of the code should be moved out of the controller script
+and into other core analysis files in this directory.
+
+The `results` directory will generally require additional subdirectories for all but the simplest projects.
+Intermediate files such as cleaned data,
+statistical tables,
+and final publication-ready figures or tables
+should be separated clearly by file naming conventions or placed into different subdirectories;
+those belonging to different papers or other publications
+should be grouped together.
+
+If the controller script(s) can automatically regenerate all of the results in a few minutes,
```

Comment:
Just to make sure I understand, you disagree with the idea that the results folder should not be placed under version control (or the results otherwise saved) since it's useful to be able to easily compare the results before and after a code change? Sounds reasonable, I can add something to that effect to the text. (Of course this is much harder to do if the results are large.)